Formula/pinot.rb

```diff
@@ -0,0 +1,35 @@
+class Pinot < Formula
+  desc "Realtime distributed OLAP datastore"
+  homepage "https://pinot.apache.org/"
+  url "https://downloads.apache.org/pinot/apache-pinot-0.9.3/apache-pinot-0.9.3-bin.tar.gz"
+  sha256 "c253eb9ce93f11f368498229282846588f478cb6e0359e24167b13e97417c025"
+  license "Apache-2.0"
+  head "https://github.com/apache/pinot.git", branch: "master"
+
+  depends_on "libtool" => :build
+
+  depends_on "openjdk"
+
+  def install
+    (var/"lib/pinot/data").mkpath
+
+    libexec.install "lib"
+    libexec.install "plugins"
+
+    prefix.install "bin"
+    bin.env_script_all_files(libexec/"bin", Language::Java.java_home_env)
+    Dir["#{bin}/*.sh"].each { |f| mv f, f.to_s.gsub(/.sh$/, "") }
+  end
+
+  service do
+    run ["/usr/local/bin/pinot-admin", "QuickStart", "-type", "BATCH", "-dataDir", var/"lib/pinot/data"]
+    keep_alive true
+    working_dir HOMEBREW_PREFIX
+    log_path var/"log/pinot/pinot_output.log"
+    error_log_path var/"log/pinot/pinot_output.log"
+  end
+
+  test do
+    assert_match("StartController", shell_output("#{bin}/pinot-admin -h 2>&1"))
```

Comment:
Can we get a better test here than calling for help or checking the version?

Reply:
Yes, for sure. I'm not experienced with Ruby so I did the shortcut here. A proper test would start the Pinot HTTP server, as it does in the service command, and wait for it to become available before executing a command to insert test data into a new table. It would then verify that the table was created in the assertion using an HTTP request. Is there an HTTP client available for tests? Also, the startup process can take about a minute, so the test would need to wait for some time before the server becomes available. Do you typically use thread sleep for that or is there a better way?

Reply:
> Is there an HTTP client available for tests?

Does `curl` work?

> Do you typically use thread sleep for that or is there a better way?

Yea, we usually just `fork` the server process and `sleep` for a little bit while it's starting up. `grep`ping for `fork do` in this repository will give you many examples.

Reply:
Thanks Carlo! I'll write the test and submit it.

Reply:
@carlocab I've created the new test method. It's slightly revised from what I originally proposed, since there would be a port conflict issue using the `pinot-admin quickstart` command that's used in the `service` method. What I did instead is to start the main HTTP service component of the recipe, which does allow a port override for itself and Zookeeper via command-line arguments. Waiting to see how the tests pan out to see whether or not the `sleep` timings need to be adjusted.